drivers/sensor/st/lis2dw12/lis2dw12.c

```diff
@@ -92,6 +92,18 @@ static inline void lis2dw12_convert(struct sensor_value *val, int raw_val,
 	val->val2 = dval % 1000000LL;
 }
 
+static inline void lis2dw12_channel_get_temp(const struct device *dev, struct sensor_value *val)
+{
+	struct lis2dw12_data *lis2dw12 = dev->data;
+	int64_t dval;
+
+	dval = ((lis2dw12->temp >> LIS2DW12_SHIFT_TEMP) * LIS2DW12_TEMP_SCALE_FACTOR);
+	dval += 25000000;
```

Comment:
This calculatio is in micro Celsius, right?
So, maybe, you may want to put a comment like that, or change variable name to dval_uc (or both).

Moreover, I would define scale factor to be 16, and use:

```* 1000000 / LIS2DW12_TEMP_SCALE_FACTOR``` 

specifying that is in micro celsius.
Would that be fine?

Reply:
I've done the scaling like this to keep only multiplication as it's more efficient this way.

I will change the variable names and add comments

Reply:
I think the compiler would optimize it anyway, but if you want to keep 62500 I think you may also want to add in the comments that it is 1000000/16.